homeassistant/components/apple_tv/config_flow.py

```diff
@@ -0,0 +1,411 @@
+        or credentials.get(Protocol.DMAP.value) is not None
+    )
+
+
+class AppleTVConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Apple TV."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get options flow for this handler."""
+        return AppleTVOptionsFlow(config_entry)
+
+    def __init__(self):
+        """Initialize a new AppleTVConfigFlow."""
+        self.target_device = None
+        self.scan_result = None
+        self.atv = None
+        self.protocol = None
+        self.pairing = None
+        self.credentials = {}  # Protocol -> credentials
+
+    async def async_step_reauth(self, info):
+        """Handle initial step when updating invalid credentials."""
+        await self.async_set_unique_id(info[CONF_IDENTIFIER])
+        self.target_device = info[CONF_IDENTIFIER]
+
+        # pylint: disable=no-member # https://github.com/PyCQA/pylint/issues/3167
+        self.context["title_placeholders"] = {"name": info[CONF_NAME]}
+
+        # pylint: disable=no-member # https://github.com/PyCQA/pylint/issues/3167
+        self.context["identifier"] = self.unique_id
+        return await self.async_step_reconfigure()
+
+    async def async_step_reconfigure(self, user_input=None):
+        """Inform user that reconfiguration is about to start."""
+        if user_input is not None:
+            return await self.async_find_device_wrapper(
+                self.async_begin_pairing, allow_exist=True
+            )
+
+        return self.async_show_form(step_id="reconfigure")
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        # Be helpful to the user and look for devices
+        if self.scan_result is None:
+            self.scan_result, _ = await device_scan(None, self.hass.loop)
+
+        errors = {}
+        default_suggestion = self._prefill_identifier()
+        if user_input is not None:
+            self.target_device = user_input[DEVICE_INPUT]
+            try:
+                await self.async_find_device()
+            except DeviceNotFound:
+                errors["base"] = "no_devices_found"
+            except DeviceAlreadyConfigured:
+                errors["base"] = "already_configured"
+            except exceptions.NoServiceError:
+                errors["base"] = "no_usable_service"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                await self.async_set_unique_id(
+                    self.atv.identifier, raise_on_progress=False
+                )
+                return await self.async_step_confirm()
+
+            # Use whatever the user entered as default value
+            default_suggestion = self.unique_id
```

Comment:
The unique id isn't the same as what the user entered in the form.

Reply:
I believe a side-effect here is that discovery will re-suggest devices that were manually configured:
![image](https://user-images.githubusercontent.com/1203111/98047788-27aef280-1df2-11eb-8b19-051b2174ac17.png)

Reply:
doc mentines that 

> Support for YAML is removed, so old configuration must be removed and devices re-added via the integrations page

Reply:
With testing, I've concluded it doesn't make a difference here. It is a manual step where an error occurred. It is fine to pass along the error device (which is None in most cases anyway). This removed the 2 lines to address the last open comment by @MartinHjelmare.

🎉